File: Queue.c
#include<stdio.h>
#include<stdlib.h>
struct node
{
    int data;
    struct node *next;
};
typedef struct node NODE;
typedef struct node* PNODE;
typedef struct node** PPNODE;

void Display(PNODE First){
     printf("Elements of Queue are : \n");
 

    while(First!= NULL)
    {
        printf("%d\n",First->data);
        First= First -> next;
    }
    
    printf("\n");
}
int Count(PNODE First){
int iCnt=0;

    while(First!= NULL)
    {
       iCnt++;

        First= First -> next;
    }
    
return iCnt;

}
void EnQueue(PPNODE First,int No){
    PNODE newn=NULL;
    PNODE temp=NULL;

    newn=(PNODE)malloc(sizeof(NODE));

    newn->data=No;
    newn->next=NULL;

    if(*First==NULL){

        *First=newn;

    }
    else{
        temp=*First;

        while((temp)->next !=NULL){

            temp=temp->next;
        }
        temp->next=newn;
        
    }
}
int DeQueue(PPNODE First){

  int iValue=0;
  PNODE temp=NULL;

  if(*First==NULL){

    printf("unable top pop the element as Queue is empty : ");

    return -1;

  }
  else{

    temp=*First;
    
    iValue=(*First)->data;
    *First=(*First)->next;
    free(temp);
  }
return iValue;
}


int main(){

PNODE Head=NULL;

int iRet=0;

EnQueue(&Head,10);
EnQueue(&Head,20);
EnQueue(&Head,30);
EnQueue(&Head,40);


Display(Head);

iRet=Count(Head);
printf("Number of elements are : %d\n",iRet);


iRet=DeQueue(&Head);
printf("Removed element is : %d\n",iRet);



iRet=DeQueue(&Head);
printf("Removed element is :%d \n",iRet);





    return 0;

}

File: Queue.java

class Queue{

    public static void main(String[] args) {
        int iRet=0;


        QueueX Aobj =new  QueueX();
        Aobj.EnQueue(11);
        Aobj.EnQueue(21);
        Aobj.EnQueue(51);
        Aobj.EnQueue(101);
        Aobj.EnQueue(121);
        Aobj.EnQueue(151);
        Aobj.Display();
        iRet=Aobj.Count();
        System.out.println("Number of elements are : "+iRet);

        Aobj.DeQueue();
        Aobj.Display();
        iRet=Aobj.Count();
        System.out.println("Number of elements are : "+iRet);


    }
}
class node{

   public int data;
   public node next;

   node(int iNo){
    
      data=iNo;
      next=null;
   }
}

class QueueX{

    public node First;
    int iCount;

    public QueueX(){

        First=null;
        iCount=0;

    }
public void EnQueue(int No){
    node newn=null;
    node temp=null;


     newn=new node(No);

    newn.data=No;
    newn.next=null;

    if(First==null){

        First=newn;
    }
    else
    {
        temp=First;

        while(temp.next!=null){

            temp=temp.next;

        }
        temp.next=newn;
      
    }
    iCount++;
}

public void Display(){

    node temp=First;

   while (temp !=null) { 
    System.out.print("|"+temp.data+"|->");
    temp=temp.next;
   }
   System.out.print("| null |\n");

  
   
}
public int Count(){

    return  iCount;

}
public void DeQueue(){

    if(First==null){
        System.out.println("The LL is empty: ");
        return;
    }
    else if(First.next==null){

        First=null;

    }
    else{

        First=First.next;


    }
    iCount--;

}


}
